### ISSUE TITLE
Date and time not updating with shouldCloseOnSelect equals false in DateRangePicker from react-aria-components

### ISSUE THREAD

**Issue by upender7270** [adobe/react-spectrum#6957]
### Provide a general summary of the issue here

I am trying to use the DateRangePicker from react-aria-components.
When the props `shouldCloseOnSelect = false`  and `granularity="minute" ` are set, then select a date range, date time not updating in the picker (DateField group)
This happened only when we had no date time set initially.

I have added the codesandbox link and video of the behaviour

### 🤔 Expected Behavior?

with `shouldCloseOnSelect = false` On select date range, then the selected date range should be visible in the trigger (FieldGroup)

### 😯 Current Behavior

with `shouldCloseOnSelect = false` On select date range, then the selected date range is not visible in the trigger (FieldGroup)

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/react-spectrum-range-picker-forked-gk44tw?workspaceId=32ded442-599e-4949-b8e5-e71ce8c0b757
[Screencast from 27-08-24 06:54:54 PM IST.webm](https://github.com/user-attachments/assets/34751c0e-60d1-492d-aec6-0f49096c0df2)

### Version

1.3.3

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Ubuntu 22.04

**Comment by LFDanLu** [comment]
This looks to be because of this if block: https://github.com/adobe/react-spectrum/blob/a899bdfb6c4020c0ef17d87fe57ab2782e895c1c/packages/%40react-stately/datepicker/src/useDateRangePickerState.ts#L134-L141
This means the DateRangePicker will only uses the placeholder time values if the operation will close the calendar popover. I feel like the following else statement isn't necessary and that we should always use the time value placeholders if none are provided. Unfortunately, this behavior seems to have been added quite intentionally (and thus might be problematic to change), trying to dig up why.

**Comment by henkkasoft** [comment]
@LFDanLu I agree this looks strange. If shouldCloseOnSelect is false the  code will never go to commitValue if timeRange is not set. Similar problem also here https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/datepicker/src/useDatePickerState.ts
Not sure how shouldCloseOnSelect with granularity='hour' or less should be used?

**Comment by bartekczyz** [comment]
Hello @LFDanLu, I'm experiencing pretty much the same problem here. It works fine if I don't control my date range picker component or if I initiate it with `null` value, but once the value is controlled and I manually `state.setValue`, it no longer calls `onChange` after clicking the "clear" button (this one clears the value).

The only workaround that comes to my mind is to `useEffect` and call `onChange` each time the value changes, are there better alternatives?

@upender7270 mate, how did you get over this?

**Comment by bartekczyz** [comment]
>  Is there a reason for your implementation not to include those time fields in the dropdown as well?

Yeah :D basically 
1. it looks insanely bad
2. we don't need that at all
3. it's a product decision

Currently I'm setting the value in `useEffect`, not the ideal solution but works
